Sources/SwiftSyntaxMacroExpansion/MacroReplacement.swift

```diff
@@ -277,6 +338,11 @@ extension MacroDeclSyntax {
           (replacement.reference, replacement.parameterIndex)
         }
       ),
+      genericReplacements: Dictionary(
+        uniqueKeysWithValues: replacements.map { replacement in
+          (replacement.reference, replacement.parameterIndex)
+        }
+      ),
```

Comment:
A user might specify `genericReplacements` when calling `expand`. So we can’t assume that the keys are indeed unique. We should fail somewhat gracefully if they are not unique.